From 974bed2e482e5ff6bb4337a5bd8d5a288aa20af4 Mon Sep 17 00:00:00 2001
From: Bjorn Helgaas <bhelgaas@google.com>
Date: Tue, 7 Mar 2023 12:19:21 -0600
Subject: [PATCH 3/4] octeon_ep: Drop redundant
 pci_enable_pcie_error_reporting()

pci_enable_pcie_error_reporting() enables the device to send ERR_*
Messages.  Since f26e58bf6f54 ("PCI/AER: Enable error reporting when AER is
native"), the PCI core does this for all devices during enumeration, so the
driver doesn't need to do it itself.

Remove the redundant pci_enable_pcie_error_reporting() call from the
driver.  Also remove the corresponding pci_disable_pcie_error_reporting()
from the driver .remove() path.

Note that this only controls ERR_* Messages from the device.  An ERR_*
Message may cause the Root Port to generate an interrupt, depending on the
AER Root Error Command register managed by the AER service driver.

Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
Cc: Veerasenareddy Burru <vburru@marvell.com>
Cc: Abhijit Ayarekar <aayarekar@marvell.com>
Acked-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Signed-off-by: Jakub Kicinski <kuba@kernel.org>
(cherry picked from commit fe3f4c292da1908eda5d322e60b3c09ed9508288)
[jma: This commit was cherry-picked from
 https://git.yoctoproject.org/linux-yocto/commit/?h=fe3f4c292da1]
Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 drivers/octeon_ep/octep_main.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/octeon_ep/octep_main.c b/drivers/octeon_ep/octep_main.c
index 675f7ab..43f388e 100644
--- a/drivers/octeon_ep/octep_main.c
+++ b/drivers/octeon_ep/octep_main.c
@@ -8,7 +8,6 @@
 #include <linux/types.h>
 #include <linux/module.h>
 #include <linux/pci.h>
-#include <linux/aer.h>
 #include <linux/netdevice.h>
 #include <linux/etherdevice.h>
 #include <linux/rtnetlink.h>
@@ -1744,7 +1743,6 @@ static int octep_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 		goto err_pci_regions;
 	}
 
-	pci_enable_pcie_error_reporting(pdev);
 	octep_enable_ptm(pdev);
 	pci_set_master(pdev);
 
@@ -1773,7 +1771,6 @@ static int octep_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 	return 0;
 
 err_alloc_netdev:
-	pci_disable_pcie_error_reporting(pdev);
 	pci_release_mem_regions(pdev);
 err_pci_regions:
 err_dma_mask:
@@ -1817,7 +1814,6 @@ static void octep_remove(struct pci_dev *pdev)
 free_resources:
 	pci_release_mem_regions(pdev);
 	free_netdev(oct->netdev);
-	pci_disable_pcie_error_reporting(pdev);
 	pci_disable_device(pdev);
 }
 
-- 
2.42.0

